src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -84,6 +107,30 @@ public void createWorld(World.Environment environment, String name) {
         }
     }
 
+    // For the regenerateTerrain() method to work correctly, we need to access the cached world, which we create here.
+    public void createCacheWorld(World world) {
+
+        if(!IridiumSkyblock.getInstance().getConfiguration().generatorType.isTerrainGenerator()) return;
+
+
+        if (Bukkit.getWorld(getCacheWorldName(world)) == null) {
+
+            WorldCreator worldCreator = new WorldCreator(getCacheWorldName(world))
+                    .generator(IridiumSkyblock.getInstance().getDefaultWorldGenerator(world.getName(), null))
+                    .environment(world.getEnvironment())
+                    .seed(world.getSeed());
+
+            worldCreator.createWorld();
+        }
+
+        if (Bukkit.getWorld(getCacheWorldName(world)).getSeed() != Bukkit.getWorld(world.getName()).getSeed()) {
+            File cacheWorld = Bukkit.getWorld(getCacheWorldName(world)).getWorldFolder();
+            Bukkit.unloadWorld(getCacheWorldName(world), false);
+            deleteWorld(cacheWorld);
```

Comment:
What happens if the server crashes or the code above throws an exception? Maybe we should cleanup cached worlds on server startup?

Reply:
we need to have the cached world for terrain generation, generating them on startup for every server start seems a bit unnecessary if the world is already there and matches our criteria